src/vanilla/GeneratorSettingsJs.cs

```diff
@@ -42,5 +51,186 @@ public class GeneratorSettingsJs : IGeneratorSettings
         /// The folder where the generated files will be output to.
         /// </summary>
         public string OutputFolder { get; set; }
+
+        /// <summary>
+        /// If the PackageVersion property is null or empty, then first try to update it from an
+        /// existing package.json file in the output directory. If no existing package.json file
+        /// exists or it doesn't contain a version property, then update it to the latest version
+        /// found on NPM.
+        /// </summary>
+        public void UpdatePackageVersion()
+        {
+            IHost host = Settings.Instance?.Host;
+            if (string.IsNullOrEmpty(PackageVersion) && host != null)
+            {
+                string[] outputFolderContents = host.ListInputs("./").Result;

```

Comment:
Is this an async call?

Reply:
It is, but there's no need to use it as an async call since our generator is very synchronous.

Reply:
I see. I assume it won't result in any deadlock? What's the environment we run it in?

Reply:
I have actually seen the generator hang in CI on rare occasions. Couldn't say whether the use of .Result is related or not.

The autorest.typescript extension gets spawned by the autorest core process. People generally run autorest on dev machines and CI environments.. Did that fully answer your question?

Reply:
Yes, thanks! :)